key.core/src/main/java/de/uka/ilkd/key/strategy/feature/AppliedRuleAppsNameCache.java

```diff
@@ -0,0 +1,126 @@
+package de.uka.ilkd.key.strategy.feature;
+
+import java.util.*;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.Nonnull;
+
+import de.uka.ilkd.key.logic.Name;
+import de.uka.ilkd.key.proof.Node;
+import de.uka.ilkd.key.rule.RuleApp;
+import de.uka.ilkd.key.util.AssertionFailure;
+
+import org.key_project.util.LRUCache;
+
+/**
+ * Establishes a cache for the applied rule apps to query them by name.
+ * See the get method for additional required constraints for correctness.
+ *
+ * @author Julian Wiesler
+ */
+public class AppliedRuleAppsNameCache {
+    /** cache of all applied rules by name of a node */
+    private final LRUCache<Node, HashMap<Name, List<RuleApp>>> cache = new LRUCache<>(32);
```

Comment:
Is the number of 32 enough. Is the performance/cache hits better with higher number?